### ISSUE TITLE
Allow short docstrings in pydoclint

### ISSUE THREAD

**Issue by valentincalomme** [astral-sh/ruff#13086]
As of `ruff==0.6.2`, `pydoclint` rules (DOC) doesn't allow any settings. If prioritization is required, I would personally really value having the `--skip-checking-short-docstrings` setting (see: https://jsh9.github.io/pydoclint/config_options.html)

This is currently blocking me from using the rule altogether.

**Comment by tjkuson** [comment]
Possibly related to #5860

**Comment by AlexWaygood** [comment]
@valentincalomme or @augustelalande, would you be able to elaborate a bit more about why this config option is desirable? I don't feel like #13302 gives a great description of what the config option is for right now, but I'm not sure how to improve it as I don't fully feel like I understand why it would be useful.

**Comment by valentincalomme** [comment]
@AlexWaygood for me it is desirable in the very simple scenario where I want to add a "one-liner" docstring, just to explain briefly what my method does. I dont' want pydoclint to complain that I am missing descriptions for the parameters. Actually, a good description is how `darglint` describes their strictness (https://pypi.org/project/darglint/). They have 3 modes:

- short: One-line descriptions are acceptable; anything more and the docstring will be fully checked.
- long: One-line descriptions and descriptions without arguments/returns/yields/etc. sections will be allowed. Anything more, and the docstring will be fully checked.
- full: (Default) Docstrings will be fully checked.
 
 Right now, the `DOC` rules are always set to "full" strictness, which forces all docstrings to include everything (parameters, returns, raises, etc.). I personally like to follow the "if you add it, it needs to be correct, but omitting it is ok" approach for docstrings

**Comment by tmke8** [comment]
I, too, always use `--skip-checking-short-docstrings` for pylint. I'm using type hints for all parameters anyway, so if it's just a simple function, I don't feel the need to document all parameters explicitly with a sentence; all I want is a one-line explanation of what the function does:

```python
def square(x: float) -> float:
    """Square the given number."""
    ...
```

I mean I guess it is a little weird to tie the requirement to document all parameters to the number of lines in the docstring, but I found this heuristic to work well.

**Comment by tmke8** [comment]
Maybe a better rule would actually be:

- If one `:param ...:` has been specified, all params and return value and exceptions need to be documented.

Instead of

- If the docstring has more than one line, all params and return value and exceptions need to be documented.

**Comment by kemus** [comment]
Hey,

Google's python style guide explicitly allows for one-liner docstrings to [omit the special sections](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) like `Args:` and `Returns:`

>Certain aspects of a function should be documented in special sections, listed below. Each section begins with a heading line, which ends with a colon. All sections other than the heading should maintain a hanging indent of two or four spaces (be consistent within a file). **These sections can be omitted in cases where the function’s name and signature are informative enough that it can be aptly described using a one-line docstring.**

(Emphasis mine). Huge amounts of google-style python code follows this pattern for simple methods.

Since we support Google's docstrings style, we should not only allow short docstrings in pydoclint, but also enable it by default if the following setting is enabled:
```
[tool.ruff.lint.pydocstyle]
convention = "google"
```

**Comment by AlexWaygood** [comment]
I tried running the pydoclint rules on my [typeshed-stats](https://github.com/AlexWaygood/typeshed-stats) project while considering whether they were ready to be stabilised for the 0.9 release. I also found that the majority of complaints were for one-line docstrings that I didn't really want to fix, but that most of the complaints regarding my multiline docstrings were helpful. So I think I'm now persuaded that this config option would indeed be useful (and it's a point in its favour that the config option also exists in the upstream tools).
